test/Field/Axioms.hs

```diff
@@ -102,24 +104,52 @@ test_mulDistributive _ =
 
 -- Other properties like  0.x == 0  and  x /= 0 => (xa == xb => a == b)  follow from the axioms.
 
-test_axioms :: TestableField f => NamedType f -> TestTree
-test_axioms namedType =
-    testGroup (coerce namedType)
-        [ test_addAssociative            namedType
-        , test_addCommutative            namedType
-        , test_zerAdditiveIdentity       namedType
-        , test_additiveInverse           namedType
-        , test_mulAssociative            namedType
-        , test_mulCommutative            namedType
-        , test_oneMultiplicativeIdentity namedType
-        , test_nonZeroInverse            namedType
-        , test_mulDistributive           namedType
+-- | Test that any 'Int' can be converted to a field element and back, resulting in the same @Int@.
+test_fromToInteger :: forall f proxy. (Field f, AsInteger f) => proxy f -> TestTree
```

Comment:
Shouldn't we use `Data.Proxy`?

Reply:
We instantiate that `proxy` with a different thing: `NamedType`. Anyway, it's like 

> Use `void` in positive position for maximum generality
> Use `Void` in negative position to clearly state your demands

(taken from [To Void or to void](https://www.fpcomplete.com/blog/2017/07/to-void-or-to-void)) except the opposite:

> Use `proxy` in negative position for maximum generality
> Use `Proxy` in positive position to clearly state your demands

(as categorically `Proxy` is a terminal object and `Void` is an initial one, hence the duality).